extensions/applicationinsights-clickanalytics-js/src/ClickAnalyticsPlugin.ts

```diff
@@ -5,19 +5,74 @@
 import {
-    BaseTelemetryPlugin, IAppInsightsCore, IConfiguration, ICustomProperties, IPlugin, IProcessTelemetryContext,
+    BaseTelemetryPlugin, IAppInsightsCore, IConfigDefaults, IConfiguration, ICustomProperties, IPlugin, IProcessTelemetryContext,
     IProcessTelemetryUnloadContext, ITelemetryItem, ITelemetryPluginChain, ITelemetryUnloadState, _eInternalMessageId, _throwInternal,
-    arrForEach, dumpObj, eLoggingSeverity, getExceptionName, isNullOrUndefined, throwError, unloadComponents
+    arrForEach, createProcessTelemetryContext, dumpObj, eLoggingSeverity, getExceptionName, isNullOrUndefined, onConfigChange, throwError,
+    unloadComponents
 } from "@microsoft/applicationinsights-core-js";
 import { PropertiesPlugin } from "@microsoft/applicationinsights-properties-js";
-import { IAutoCaptureHandler, IClickAnalyticsConfiguration, IContentHandler, IPageActionTelemetry } from "./Interfaces/Datamodel";
-import { BehaviorEnumValidator, BehaviorMapValidator, BehaviorValueValidator, mergeConfig } from "./common/Utils";
+import { hasDocument, objDeepFreeze, objDefineProp, objForEachKey } from "@nevware21/ts-utils";
+import {
+    IAutoCaptureHandler, IClickAnalyticsConfiguration, IContentHandler, ICoreData, ICustomDataTags, IPageActionTelemetry
+} from "./Interfaces/Datamodel";
+import {
+    BehaviorEnumValidator, BehaviorMapValidator, BehaviorValueValidator, DEFAULT_AI_BLOB_ATTRIBUTE_TAG, DEFAULT_DATA_PREFIX,
+    DEFAULT_DONOT_TRACK_TAG, removeInvalidElements
+} from "./common/Utils";
 import { PageAction } from "./events/PageAction";
 import { AutoCaptureHandler } from "./handlers/AutoCaptureHandler";
 import { DomContentHandler } from "./handlers/DomContentHandler";
 
 export { BehaviorMapValidator, BehaviorValueValidator, BehaviorEnumValidator }
 
+const dataTagsDefault = {
+    useDefaultContentNameOrId: false,
+    aiBlobAttributeTag: DEFAULT_AI_BLOB_ATTRIBUTE_TAG,
+    customDataPrefix: DEFAULT_DATA_PREFIX,
+    captureAllMetaDataContent: false,
+    dntDataTag: DEFAULT_DONOT_TRACK_TAG,
+    metaDataPrefix: "",
+    parentDataTag: ""
+} as ICustomDataTags;
+
+const coreDataDefault = {
+    eferrerUri: hasDocument ? document.referrer : "",
+    requestUri: "",
+    pageName: "",
+    pageType: ""
+} as ICoreData;
+
+const defaultValues: IConfigDefaults<IClickAnalyticsConfiguration> = objDeepFreeze({
+    autoCapture: true,
+    callback: {
+        pageActionPageTags: () => null,
+        pageName: (element?: Element) => null,
+        contentName: (element?: any, useDefaultContentName?: boolean) => null
+    },
+    pageTags: {},
+    coreData: {set:_setProp, v:coreDataDefault},
+    dataTags: {set: _setProp, v: dataTagsDefault},
+    behaviorValidator: (key:string) => key || "",
+    defaultRightClickBhvr: "",
+    dropInvalidEvents : false,
+    urlCollectHash: false,
+    urlCollectQuery: false
+});
+
+function _setProp(val: Object, def: Object): Object {
+    removeInvalidElements(val);
+    if (JSON.stringify(def) !== "{}") {
```

Comment:
This would better being either `if (def) {` or `if (isObject(def)) {` as `JSON.stringify()` can be expensive.